Sources/Auth/Services/App/AuthRespondSubscriber.swift

```diff
@@ -7,30 +7,48 @@ class AuthRespondSubscriber {
     private let networkingInteractor: NetworkInteracting
     private let logger: ConsoleLogging
     private let rpcHistory: RPCHistory
+    private let signatureVerifier: MessageSignatureVerifying
+    private let messageFormatter: SIWEMessageFormatting
     private var publishers = [AnyCancellable]()
     var onResponse: ((_ id: RPCID, _ cacao: Cacao) -> Void)?
 
     init(networkingInteractor: NetworkInteracting,
          logger: ConsoleLogging,
-         rpcHistory: RPCHistory) {
+         rpcHistory: RPCHistory,
+         signatureVerifier: MessageSignatureVerifying,
+         messageFormatter: SIWEMessageFormatting) {
         self.networkingInteractor = networkingInteractor
         self.logger = logger
         self.rpcHistory = rpcHistory
+        self.signatureVerifier = signatureVerifier
+        self.messageFormatter = messageFormatter
         subscribeForResponse()
     }
 
     private func subscribeForResponse() {
         networkingInteractor.responsePublisher.sink { [unowned self] subscriptionPayload in
-            guard let request = rpcHistory.get(recordId: subscriptionPayload.request.id!)?.request,
-                  request.method == "wc_authRequest" else { return }
+            guard
+                let requestId = subscriptionPayload.request.id,
+                let request = rpcHistory.get(recordId: requestId)?.request,
+                request.method == "wc_authRequest" else { return }
+
             networkingInteractor.unsubscribe(topic: subscriptionPayload.topic)
-            guard let cacao = try? subscriptionPayload.request.result?.get(Cacao.self) else {
-                logger.debug("Malformed auth response params")
-                return
-            }
+
             do {
-                try CacaoSignatureVerifier().verifySignature(cacao)
-                onResponse?(subscriptionPayload.request.id!, cacao)
+                guard let cacao = try subscriptionPayload.request.result?.get(Cacao.self) else {
+                    return logger.debug("Malformed auth response params")
+                }
+
+                let address = try DIDPKH(iss: cacao.payload.iss).account.address
+                let payload = AuthPayload(payload: cacao.payload)
```

Comment:
instead of initialising `AuthPayload` from cacao I would rather add new method to messageFormatter:
`messageFormatter.formatMessage(from: cacao)`